lib/segment/src/segment.rs

```diff
@@ -658,6 +658,15 @@ impl Segment {
     pub fn total_point_count(&self) -> usize {
         self.id_tracker.borrow().total_point_count()
     }
+
+    pub fn preheat_disk_cache(&self) -> impl Iterator<Item = mmap_ops::PreheatDiskCache> + '_ {
+        self.vector_data
+            .values()
+            .filter_map(|storage| match &*storage.vector_storage.borrow() {
+                VectorStorageEnum::Memmap(storage) => storage.preheat_disk_cache(),
+                _ => None,
```

Comment:
We probably want to preheat `AppendableMemmap` here as well.

```suggestion
                VectorStorageEnum::Memmap(storage) => storage.preheat_disk_cache(),
                VectorStorageEnum::AppendableMemmap(storage) => storage.preheat_disk_cache(),
                VectorStorageEnum::Simple(_) => None,
```

I think keeping this exhaustive is good.

Reply:
Additionally, we might have mmap in HNSW

Reply:
Didn't implement it for `AppendableMmapVectorStorage`/`ChunkedMmapVectors` (yet) presicely because it contains `MmapMut` and was unsure how to proceed with them. The current code structure we have, __kinda__ allow us to implement these, but I'm already not sure regarding correctness of how we use `MmapMut`. :/